tf/tfprocess.py

```diff
@@ -215,10 +220,21 @@ def process(self, batch_size, test_batches):
             self.calculate_test_summaries(test_batches, steps + 1)
 
         # Run training for this batch
-        policy_loss, mse_loss, reg_term, _, _ = self.session.run(
-            [self.policy_loss, self.mse_loss, self.reg_term, self.train_op,
-                self.next_batch],
-            feed_dict={self.training: True, self.learning_rate: self.lr, self.handle: self.train_handle})
+        self.session.run(self.zero_op)
+        for _ in range(batch_splits):
+            policy_loss, mse_loss, reg_term, _, _ = self.session.run(
+                [self.policy_loss, self.mse_loss, self.reg_term, self.accum_op,
+                    self.next_batch],
+                feed_dict={self.training: True, self.handle: self.train_handle})
+            # Keep running averages
+            # Google's paper scales MSE by 1/4 to a [0, 1] range, so do the same to
+            # get comparable values.
+            mse_loss /= 4.0
+            self.avg_policy_loss.append(policy_loss)
+            self.avg_mse_loss.append(mse_loss)
+            self.avg_reg_term.append(reg_term)
+        self.session.run(self.train_op,
+            feed_dict={self.learning_rate: self.lr / batch_splits, self.training: True, self.handle: self.train_handle})
```

Comment:
This is possibly worth a comment that self.lr / batch_splits is used because the gradient has been accumulated, not averaged.